test/suites/dev/test-xcm/test-xcm-erc20-fees-and-trap.ts

```diff
@@ -0,0 +1,357 @@
+        expectEVMResult(result!.events, "Succeed");
+
+        // Check the sovereign account has received erc20 tokens
+        expect(
+          await context.readContract!({
+            contractName: "ERC20WithInitialSupply",
+            contractAddress: erc20ContractAddress as `0x${string}`,
+            functionName: "balanceOf",
+            args: [paraSovereign],
+          })
+        ).equals(amountTransferred);
+
+        // Create xcm message to send ERC20 tokens to Charleth
+        // We don't buy any execution with native currency
+        const config: XcmFragmentConfig = {
+          assets: [
+            {
+              multilocation: {
+                parents: 0,
+                interior: {
+                  X2: [
+                    {
+                      PalletInstance: erc20XcmPalletIndex,
+                    },
+                    {
+                      AccountKey20: {
+                        network: "Any",
+                        key: stringToU8a(erc20ContractAddress),
+                      },
+                    },
+                  ],
+                },
+              },
+              fungible: amountTransferred,
+            },
+          ],
+          beneficiary: CHARLETH_ADDRESS,
+        };
+
+        // Build the xcm message
+        const xcmMessage = new XcmFragment(config)
+          .withdraw_asset()
+          .clear_origin()
+          .buy_execution()
+          .deposit_asset(2n)
+          .as_v2();
+
+        // Mock the reception of the xcm message
+        await injectHrmpMessage(context, paraId, {
+          type: "XcmVersionedXcm",
+          payload: xcmMessage,
+        });
+        await context.createBlock();
+
+        // Search for Fail event
+        const records = await polkadotJs.query.system.events();
+        const events = records.filter(
+          ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+        );
+
+        // Check the error is TooExpensive
+        expect(events).to.have.lengthOf(1);
+        expect((events[0].event.data.toHuman() as any).error).equals("TooExpensive");
+
+        // Charleth should not receive ERC20 tokens due to failed execution
+        expect(
+          await context.readContract!({
+            contractName: "ERC20WithInitialSupply",
+            contractAddress: erc20ContractAddress as `0x${string}`,
+            functionName: "balanceOf",
+            args: [CHARLETH_ADDRESS],
+          })
+        ).equals(0n);
+      },
+    });
+
+    it({
+      id: "T02",
+      title: "Should not trap any ERC20",
+      test: async function () {
+        const paraId = 888;
+        const paraSovereign = sovereignAccountOfSibling(context, paraId);
+        const amountTransferred = 1_000_000n;
+
+        // Get pallet index
+        const metadata = await polkadotJs.rpc.state.getMetadata();
+        const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
```

Comment:
It would be safer to not use toHuman, and instead interact with the SCALE objects directly. e.g.
```
const balancesPalletIndex = metadata.asLatest.pallets
          .find(({ name }) => name.toString() == "Balances")!
          .index.toNumber();
```

Reply:
Applies to the indices below too